spec/API_specification/array_api/fourier_transform_functions.py

```diff
@@ -0,0 +1,512 @@
+    n: int
+        length of the transformed axis of the **output**. If
+
+        - ``n//2+1`` is greater than the length of the input array, the input array is zero-padded to length ``n//2+1``.
+        - ``n//2+1`` is less than the length of the input array, the input array is trimmed to length ``n//2+1``.
+        - ``n`` is not provided, the length of the transformed axis of the output must equal the length ``2*(m-1)``, where ``m`` is the length of the input along the axis specified by ``axis``.
+        
+        Default: ``None``.
+    axis: int
+        axis (dimension) over which to compute the Fourier transform. If not set, the last axis (dimension) is used.
+        
+        Default: ``-1``.
+    norm: Literal['backward', 'ortho', 'forward']
+        normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: no normalization.
+        - ``'ortho'``: normalize by ``1/sqrt(n)`` (i.e., make the FFT orthonormal).
+        - ``'forward'``: normalize by ``1/n``.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        an array transformed along the axis (dimension) indicated by ``axis``. The returned array must have a real-valued floating-point data type determined by :ref:`type-promotion`.
+    """
+
+
+def ihfft(x: array, /, *, n: Optional[int] = None, axis: int = -1, norm: Literal['backward', 'ortho', 'forward'] = 'backward') -> array:
+    """
+    Computes the one-dimensional inverse discrete Fourier transform of a signal with Hermitian symmetry.
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a real-valued floating-point data type.
+    n: int
+        length of the transformed axis of the **input**. If
+
+        - ``n`` is greater than the length of the input array, the input array is zero-padded to length ``n``.
+        - ``n`` is less than the length of the input array, the input array is trimmed to length ``n``.
+        - ``n`` is not provided, the length of the transformed axis of the output must equal the length of the input along the axis specified by ``axis``.
+        
+        Default: ``None``.
+    axis: int
+        axis (dimension) over which to compute the Fourier transform. If not set, the last axis (dimension) is used.
+        
+        Default: ``-1``.
+    norm: Literal['backward', 'ortho', 'forward']
+        normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: normalize by ``1/n``.
+        - ``'ortho'``: normalize by ``1/sqrt(n)`` (i.e., make the FFT orthonormal).
+        - ``'forward'``: no normalization.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        an array transformed along the axis (dimension) indicated by ``axis``. The returned array must have a complex-valued floating-point data type determined by :ref:`type-promotion`.
+    """
+
+
+def fftfreq(n: int, /, *, d: float = 1.0):
```

Comment:
It looks `-> array` is missing in the signatures of `fftfreq`, `rfftfreq`, `fftshift`, `ifftshift`.

Reply:
Thanks, @toslunar, fixed in e8e0f1b.